public/app/AppWrapper.tsx

```diff
@@ -102,11 +103,11 @@ export class AppWrapper extends React.Component<AppWrapperProps, AppWrapperState
                 actions={[]}
                 options={{ enableHistory: true, callbacks: { onSelectAction: commandPaletteActionSelected } }}
               >
-                <ModalsProvider>
-                  <GlobalStyles />
-                  <div className="grafana-app">
-                    <Router history={locationService.getHistory()}>
-                      <CompatRouter>
+                <Router history={locationService.getHistory()}>
+                  <CompatRouter>
+                    <ModalsContextProvider>
+                      <GlobalStyles />
+                      <div className="grafana-app">
```

Comment:
only change here is that Router is moved up a a bit so it's above ModalsProvider and ModalsRoot